notebooks/utils/estimation.py

```diff
@@ -0,0 +1,229 @@
+            SSVEP covariance matrices for each trials.
+        """
+        filtered = []
+        for filter in self.filters:
+            filtered_X = filter(X, axis=-1)
+            filtered.append(filtered_X)
+        mega_X = np.concatenate(filtered, axis=1)
+        out = covariances(mega_X, estimator=self._estimator)
+        return out
+
+    def _design_filters(self, fs: float, flickering_frequencies: tuple,
+                        bandwidth: float = 4, order: int = 2,
+                        filter_design: str = 'iir', padlen: int = None):
+        self.filters = []
+        for flickering_frequency in flickering_frequencies:
+            frequencies = [flickering_frequency - bandwidth / 2,
+                           flickering_frequency + bandwidth / 2]  # TODO/Question : shall we consider harmonics as well ?
+            filter_args = design_filter(frequencies=frequencies, order=order, fs=fs, filter_design=filter_design)
+            filter_func = functools.partial(filtfilt, *filter_args, padlen=padlen)
+            self.filters.append(filter_func)
+
+
+class SSVEPPotato(BaseEstimator, TransformerMixin):
+    """Estimate potato field per class
+        Parameters
+        ----------
+        classes: classes to consider in potato field (one potato per class)
+        potato_threshold: threshold for each potato
+        field_threshold: threshold for the potato field
+        n_iter_max: max iteration allowed for potato convergence
+        pos_label: label for accepted trials in predict
+        neg_label: label for rejected trials in predict
+    """
+
+    def __init__(self, classes: tuple = None, potato_threshold: float = .1, field_threshold: float = .1,
+                 n_iter_max: int = 100, pos_label=1, neg_label=0):
+
+        self.classes = classes
+        self.potato_threshold = potato_threshold
+        self.field_threshold = field_threshold
+        self.n_iter_max = n_iter_max
+        self.pos_label = pos_label
+        self.neg_label = neg_label
+
+    def fit(self, X: np.array, y: np.array) -> object:
+        """Fit.
+
+        Do nothing. For compatibility purpose.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_samples)
+            ndarray of trials.
+        y : ndarray shape (n_trials,)
+            labels corresponding to each trial, not used.
+
+        Returns
+        -------
+        self : SSVEPPotato instance
+            The SSVEPPotato instance.
+        """
+        self.classes = self.classes or set(y)
+        self._n_potatoes = len(self.classes)
+        self._potatoes = []
+        for _class in self.classes:
+            X_class = X[y == _class, :, :]
+            potato = Potato(n_iter_max=self.n_iter_max, threshold=self.potato_threshold).fit(X_class)
+            self._potatoes.append(potato)
+
+        return self
+
+    def transform(self, X: np.array) -> np.array:
```

Comment:
Minor comment: IMO, `transform()` should only collect zscores of potatoes. The combination should be made in `predict()`.